service/topology_coordinator.cc

```diff
@@ -1292,6 +1291,37 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         co_await update_topology_state(std::move(guard), std::move(updates), "Finished tablet migration");
     }
 
+    future<> handle_tablet_split_finalize(group0_guard g) {
+        std::vector<canonical_mutation> updates;
+        updates.reserve(_tables_to_finalize_split.size() * 2 + 1);
+
+        // Executes a global barrier to guarantee that any process (e.g. repair) holding stale version
+        // of token metadata will complete before we update topology.
+        auto guard = co_await global_tablet_token_metadata_barrier(std::move(g));
+
+        for (auto& table_id : _tables_to_finalize_split) {
+            auto s = _db.find_schema(table_id);
```

Comment:
If table was dropped, this will throw, and coordinator will enter exception live-lock. Can't happen if we get the decisions under the same group0  guard here.